apps/web/components/booking/RecurringBookingsTooltip.tsx

```diff
@@ -0,0 +1,87 @@
+import dayjs from "@calcom/dayjs";
+import classNames from "@calcom/lib/classNames";
+import { formatTime } from "@calcom/lib/date-fns";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { getEveryFreqFor } from "@calcom/lib/recurringStrings";
+import { BookingStatus } from "@calcom/prisma/enums";
+import { Tooltip } from "@calcom/ui";
+import { RefreshCcw } from "@calcom/ui/components/icon";
+
+import type { BookingItemProps } from "@components/booking/BookingListItem";
+
+interface RecurringBookingsTooltipProps {
+  booking: BookingItemProps;
+  recurringDates: Date[];
+  userTimeZone: string | undefined;
+  userTimeFormat: number | null | undefined;
+}
+
+const RecurringBookingsTooltip = ({
+  booking,
+  recurringDates,
+  userTimeZone,
+  userTimeFormat,
+}: RecurringBookingsTooltipProps) => {
+  const {
+    t,
+    i18n: { language },
+  } = useLocale();
+  const now = new Date();
+  const recurringCount = recurringDates.filter((recurringDate) => {
+    return (
+      recurringDate >= now &&
+      !booking.recurringInfo?.bookings[BookingStatus.CANCELLED]
+        .map((date) => date.toString())
+        .includes(recurringDate.toString())
+    );
+  }).length;
+
+  return (
+    (booking.recurringInfo &&
+      booking.eventType?.recurringEvent?.freq &&
+      (booking.listingStatus === "recurring" ||
+        booking.listingStatus === "unconfirmed" ||
+        booking.listingStatus === "cancelled") && (
```

Comment:
Can we early return if this conditional is not met? 

Havent tested but something like 

``` 
  const isRecurringBooking = booking.recurringInfo && booking.eventType?.recurringEvent;
  const isRecurringListingStatus = ["recurring", "unconfirmed", "cancelled"].includes(booking.listingStatus);
  if (
    !isRecurringBooking ||
    !isRecurringListingStatus ||
    !booking.eventType.recurringEvent ||
    !booking.recurringInfo?.count
  ) {
    return null;
  }
``` 
should work